tools/buildsteps/win32/buildhelpers.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+MAKEFLAGS=""
+BGPROCESSFILE="$2"
+targetBuild="$3"
+bits="$4"
+tools="$5"
+
+cpuCount=1
+if [ $NUMBER_OF_PROCESSORS > 1 ]; then
+  if [ $NUMBER_OF_PROCESSORS > 4 ]; then
+    cpuCount=6
+  else
+    cpuCount=`expr $NUMBER_OF_PROCESSORS + $NUMBER_OF_PROCESSORS / 2`
+  fi
+fi
+
+do_wget() {
+  local URL="$1"
+  local archive="$2"
+
+  if [[ -z $archive ]]; then
+    wget --tries=20 --retry-connrefused --waitretry=2 --no-check-certificate -c $URL
+  else
+    wget --tries=20 --retry-connrefused --waitretry=2 --no-check-certificate -c $URL -O $archive
+  fi
+}
+
+do_makeinstall() {
+  make -j $cpuCount "$@"
+  make install
+}
+
+do_print_status() {
+  local pad=$(printf '%0.1s' "."{1..40})
+  local padlen=$((${#pad}-${#1}-${#3}))
+  printf '%s %*.*s%s%s%s\n' "${bold_color}$1${reset_color}" 0 "$padlen" "$pad" " [$2" "$3" "$reset_color]"
+}
+
+do_pkgConfig() {
+  local pkg=${1%% *}
+  local version=$2
+  [[ -z "$version" ]] && version="${1##*= }"
+  [[ "$version" = "$1" ]] && version="" || version=" $version"
+  local prefix=$(pkg-config --variable=prefix --silence-errors "$1")
+  [[ ! -z "$prefix" ]] && prefix="$(cygpath -u "$prefix")"
+  if [[ "$prefix" = "$LOCALDESTDIR" || "$prefix" = "/trunk${LOCALDESTDIR}" ]]; then
+    do_print_status "${pkg}${version}" "$green_color" "Up-to-date"
+    return 1
+  fi
+}
+
+get_last_version() {
+  local filelist="$1"
+  local filter="$2"
+  local version="$3"
+  local ret=$(echo "$filelist" | /usr/bin/grep -E "$filter" | sort -V | tail -1)
+  if [[ -z "$version" ]]; then
+    echo $ret
+  else
+    echo $ret | /usr/bin/grep -oP "$version"
+  fi
+}
+
+do_autoreconf() {
+  if [[ ! -f configure ]]; then 
+    autoreconf -fiv
+  fi
+}
+
+do_clean() {
+  if [[ "$1" == "clean" ]] || [[ ! -f $ARCHIVE ]]; then
+    if [ -d $LIBNAME ]; then
+      echo " removing '$LIBNAME'"
+      rm -r $LIBNAME
+    fi
+  fi
+}
+
+do_download() {
+  if [ ! -d $LIBNAME ]; then
+    if [ ! -f $ARCHIVE ]; then
+      echo " downloading '$ARCHIVE'"
+      do_wget $BASE_URL/$VERSION.tar.gz $ARCHIVE
+    fi
+
+    echo " extracting '$ARCHIVE'"
+    mkdir $LIBNAME && cd $LIBNAME
+    tar -xaf ../$ARCHIVE --strip 1
+  fi
+}
+
+do_loaddeps() {
+  local file="$1"
+  LIBNAME=$(grep "LIBNAME=" $file | sed 's/LIBNAME=//g')
+  BASE_URL=$(grep "BASE_URL=" $file | sed 's/BASE_URL=//g')
+  VERSION=$(grep "VERSION=" $file | sed 's/VERSION=//g')
+  GNUTLS_VER=$(grep "GNUTLS_VER=" $file | sed 's/GNUTLS_VER=//g')
```

Comment:
just noticed that those sed's fail miserably when there are comments in the version file.
sed 's/VERSION=//g;s/#.*$//g;/^$/d' makes it much more robust. Can you add this here as well? I'll fix it for autotools.